syntax = "proto3";
option go_package = "go-server/calculator/protobuf";

package calculator;

service Calculator {
  rpc AddNumbers (CalculationRequest) returns (stream CalculationResponse);
  rpc MultiplyNumbers (CalculationRequest) returns (stream CalculationResponse);
}

message CalculationRequest {
  int32 number1 = 1;
  int32 number2 = 2;
}

message CalculationResponse {
  int32 result = 1;
}